src/report/report.py

```diff
@@ -0,0 +1,151 @@
+"""
+Module containing report generation.
+"""
+
+from sys import argv
+from os import remove
+
+from docx.enum import section, text
+from docx.shared import RGBColor
+
+import report_utils as utils
+import report_charts as charts
+from report_data import get_report_data
+
+
+def generate(directory: str = '') -> str:
```

Comment:
Shouldn't it be splitted into smaller functions?

Reply:
not yet resolved

Reply:
No it shouldn't. It's a function that generates this particular document. It serves one purpose only. Splitting it into smaller functions will result in functions that are used once and serve no other purpose.

Reply:
No it shouldn't. It's a function that generates this particular document. It serves one purpose only. Splitting it into smaller functions will result in functions that are used once and serve no other purpose. I've already created functions that can be reused.

Reply:
This function is far to long, without any sensible labes detailing steps of execution and thus making review process more difficult